### PR TITLE
Add port of FISTA and NAG from JAXopt

### LINKED ISSUE #380: Replace `OptaxOptimistixProximalGradient` with a port of `jaxopt.ProximalGradient`
As Optimistix doesn't implement `ProximalGradient`, this is currently set to use a custom implementation in `OptaxOptimistixProximalGradient` based on Optax's SGD with linesearch followed by the proximal operator.
This seems to work in practice, but is not as theoretically sound as `jaxopt.ProximalGradient` which implements the FISTA algorithm.

To be able to eventually stop vendoring (and drop support for) JAXopt, we decided to add a port of JAXopt's `ProximalGradient` that adheres to Optimistix's interface, and replace `OptaxOptimistixProximalGradient` with this when using the Optimistix backend.

### PR DESCRIPTION
**Pull request by bagibence**
As Optimistix doesn't implement `ProximalGradient`, this is currently set to use a custom implementation in `OptaxOptimistixProximalGradient` based on Optax's SGD with linesearch followed by the proximal operator.
This seems to work in practice, but doesn't support the original Nesterov acceleration and is not as theoretically sound as `jaxopt.ProximalGradient` which implements the FISTA algorithm.

This PR adds `OptimistixFISTA`, a port of JAXopt's `ProximalGradient` that adheres to Optimistix's interface, and replaces `OptaxOptimistixProximalGradient` with this when using the Optimistix backend.

Similarly, to support the original Nesterov acceleration in gradient descent, a short modification of this as `OptimistixNAG` is added. NAG stands for Nesterov's Accelerated Gradient.

Additional changes:
- Remove `OptimistixOptaxProximalGradient`
- Remove `stateful_scale_by_learning_rate` (and related code) which was only needed for `OptimistixOptaxProximalGradient`
- Switch `OptimistixOptaxGradientDescent` to use `optax.scale_by_backtracking_linesearch`

`OptimistixOptaxGradientDescent` is kept but is replaced in the tests. Ideally, it would be tested the same way as `OptimistixNAG`.
@BalzaniEdoardo, @billbrod is that fine with you, or is a way to include this in the tests without unnecessarily rerunning others required?

Fixes #380

### CODE REVIEW

File: src/nemos/solvers/_fista_port.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
+"""Adaptation of JAXopt's ProximalGradient (FISTA) as an Optimistix IterativeSolver."""
```
Comment (BalzaniEdoardo) [inline src/nemos/solvers/_fista_port.py]: I would rename this as _fista_linesearch.py or something like that. True that the algorithm is inspired by the jaxopt implementation but you did change enough of it to match the optimistix API that we can omit the fact that it is a port.

File: tests/conftest.py [hunk 2]
```diff
@@ -1128,8 +1128,9 @@ def population_negativeBinomialGLM_model_instantiation_pytree(
     },
     "optimistix": {
         **_common_solvers,
-        "GradientDescent": nmo.solvers.OptimistixOptaxGradientDescent,
-        "ProximalGradient": nmo.solvers.OptimistixOptaxProximalGradient,
+        # TODO: OptaxOptimistixGradientDescent is not tested
```
Comment (BalzaniEdoardo) [inline tests/conftest.py]: If I understand correctly, the default is using OptimistixNAG but we could still use the Optax GD if one wants to. 

One way to test this is to use unittest mock to mock the solver registry in the test:

```python
from unittest.mock import patch


# In your test
with patch.dict('nemos.solvers.solver_registry', 
                {'GradientDescent': nmo.solvers.OptimistixOptaxGradientDescent}):
    model = Model(solver_name='GradientDescent2', ...)
    model.instantiate_solver()
    
    assert isinstance(model._solver, nmo.solvers.OptimistixOptaxGradientDescent)
```

File: src/nemos/solvers/_fista_port.py (hunk 1 above)
Comment (bagibence) [reply src/nemos/solvers/_fista_port.py]: I renamed it to `_fista.py` and rephrased the module's docstring, keeping "Adapted from JAXopt." to credit them.

File: tests/conftest.py (hunk 2 above)
Comment (bagibence) [reply tests/conftest.py]: The problem with that is that it would have to be done for every test that uses gradient descent.
I added a solution where only tests referencing `"GradientDescent"` explicitly are rerun with the `OptimistixOptaxGradientDescent` implementation and added it as a step to the CI job for the Optimistix backend.

Comment (bagibence) [reply tests/conftest.py]: Changed this so that it is now controlled by `pytest -o override_solver=GradientDescent:OptimistixOptaxGradientDescent` and testing both implementations is done in the same tox environment.

**Review by BalzaniEdoardo — APPROVED** [review]
LGTM
